pkg/git/sshAuthHelper.go

```diff
@@ -0,0 +1,53 @@
+package git
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+)
+
+var keyscan string
```

Comment:
I think this flow is a bit brittle as it relies on global ENV_VARs and variables. 

If we want to reimplement the equivalent of `ssh-keyscan` here, we can set a`HostKeyCallback` 
from golang.org/x/crypto's `FixedHostKey`. 

Or perhaps we don't really care and should just provide a nop HostKeyCallback.

Reply:
Tekton allows git secrets to configure a `known_hosts` which would allow us to make this more secure. https://github.com/tektoncd/pipeline/blob/master/docs/auth.md

Reply:
That is a really nice way to actually set this up. Not gonna lie. Given it took me most of yesterday to figure out how to configure known hosts, I am equal parts relieved that we can just ask for that information and annoyed that my "genius" can be deleted. BIG win overall though.

Reply:
> bit brittle as it relies on global ENV_VARs and variables.

I agree. I am surprised you didn't catch the even greater sin I committed by created temp files for them and not deleting them. This PR was like a decent first pass at a solution that worked but I definitely wanted to revisit parts of it with a clear head.